components/blitz/src/omero/gateway/facility/TablesFacility.java

```diff
@@ -0,0 +1,527 @@
+            if (!sr.areTablesEnabled()) {
+                throw new Exception("Tables feature is not enabled on this server!");
+            }
+            long repId = sr.repositories().descriptions.get(0).getId()
+                    .getValue();
+            TablePrx table = sr.newTable(repId, name);
+            table.initialize(columns);
+            table.addData(columns);
+            table.close();
+
+            DataManagerFacility dm = gateway
+                    .getFacility(DataManagerFacility.class);
+            BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+
+            OriginalFile file = table.getOriginalFile();
+            file = (OriginalFile) browse.findIObject(ctx, file);
+
+            FileAnnotation anno = new FileAnnotationI();
+            anno.setFile(file);
+            FileAnnotationData annotation = new FileAnnotationData(anno);
+            annotation.setDescription(name);
+
+            annotation = (FileAnnotationData) dm.saveAndReturnObject(ctx,
+                    annotation);
+            dm.attachAnnotation(ctx, annotation, target);
+        } catch (Exception e) {
+            handleException(this, e, "Could not add table");
+        }
+    }
+
+    /**
+     * Load the data from a table
+     * 
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @param fileId
+     *            The if of the {@link OriginalFile} which stores the table
+     * @return All data which the table contains
+     * @throws DSOutOfServiceException
+     * @throws DSAccessException
+     */
+    public TableData getTable(SecurityContext ctx, long fileId)
+            throws DSOutOfServiceException, DSAccessException {
+        try {
+            OriginalFile file = new OriginalFileI(fileId, false);
+            SharedResourcesPrx sr = gateway.getSharedResources(ctx);
+            TablePrx table = sr.openTable(file);
+
+            Column[] columns = table.getHeaders();
+
+            String[] header = new String[columns.length];
+            String[] descriptions = new String[columns.length];
+            Class<?>[] types = new Class<?>[columns.length];
+            long[] colNumbers = new long[columns.length];
+            for (int i = 0; i < columns.length; i++) {
+                header[i] = columns[i].name;
+                descriptions[i] = columns[i].description;
+                colNumbers[i] = i;
+            }
+
+            if (table.getNumberOfRows() == 0) {
+                for (int i = 0; i < columns.length; i++) {
+                    types[i] = Object.class;
+                }
+                return new TableData(header, descriptions, types,
+                        new Object[columns.length][0]);
+            }
+
+            int nRows = (int) table.getNumberOfRows();
+            Object[][] dataArray = new Object[columns.length][nRows];
+            Data data = table.read(colNumbers, 0, table.getNumberOfRows());
```

Comment:
@dominikl : but in terms of API design, we should really avoid all definitions of non-paging methods. One alternative would be to just define that a given method will default to only returning N rows. (This is what `bin/omero hql` does) IQuery defaulted to returning everything in the early days and it's not yet been deprecated despite the problems it causes.